influxdb3/write.go

```diff
@@ -163,19 +158,49 @@ func (c *Client) write(ctx context.Context, buff []byte, options *WriteOptions)
 	body = bytes.NewReader(buff)
 	headers := http.Header{"Content-Type": {"application/json"}}
 	if gzipThreshold > 0 && len(buff) >= gzipThreshold {
-		body, err = gzip.CompressWithGzip(body)
+		r, err := gzip.CompressWithGzip(body)
 		if err != nil {
-			return fmt.Errorf("unable to compress write body: %w", err)
+			return nil, fmt.Errorf("unable to body gzip reader: %w", err)
```

Comment:
I feel like error message should be reworded like eg
```suggestion
			return nil, fmt.Errorf("unable to compress body: %w", err)
```